ts/src/index.ts

```diff
@@ -155,19 +155,49 @@ export async function startLocal(config: Config = {}): Promise<void> {
   const normalizedConfig = await initConfig(config);
   profiler = new Profiler(normalizedConfig);
 
+  // Set up periodic logging.
+  const logger = new common.logger({
+    level: common.logger.LEVELS[normalizedConfig.logLevel],
+    tag: pjson.name
+  });
+  let heapProfileCount = 0;
+  let timeProfileCount = 0;
+  let prevLogTime = Date.now();
+
+  setInterval(() => {
+    const curTime = Date.now();
+    const {rss, heapTotal, heapUsed} = process.memoryUsage();
+    logger.debug(
+        new Date().toISOString(), 'rss', rss, 'B', 'heapTotal', heapTotal, 'B',
+        'heapUsed', heapUsed, 'B');
+    logger.debug(
+        new Date().toISOString(), heapProfileCount,
+        'heap profiles collected in', curTime - prevLogTime, 'ms');
+    logger.debug(
+        new Date().toISOString(), timeProfileCount,
+        'time profiles collected in', curTime - prevLogTime, 'ms');
+
+    heapProfileCount = 0;
+    timeProfileCount = 0;
+    prevLogTime = curTime;
+  }, 10000);
+
+
   while (true) {
     if (!config.disableHeap) {
       const heap = await profiler.profile(
```

Comment:
You're not changing this here, but it seems like currently we collect a Heap and then a Time profile serially but in a dependent fashion. Is this intentional? The duration between two successive heap profiles would be `localPauseMillis + ` localTimeDurationMillis`. The time profiler will never be able to profile the heap profiler. This might not be optimal.

Do we want two independent intervals for each of these profilers?

Reply:
Running the two profilers sequentially is intentional - the startLocal thing is intended to emulate what the API-curated mode does, just in a much more frequent succession. Profiling heap profiler with time profiler would not be what the main flow of the profiler does, and is not too useful anyway since most of the interesting stuff happens in the heap profiler in native code.

It would make sense though to wait for the specified duration after each profile though. Maybe wait for (pause + timeDuration) after heap profile collection so that the start points of each profile are equally distanced.

Reply:
I switched to using setInterval rather than a while loop, which will, I think, provide the best assurances that profiles are collected at a specific rate.